ovos_ww_plugin_openwakeword/__init__.py

```diff
@@ -48,12 +48,12 @@ def update(self, chunk):
         audio_frame = np.frombuffer(chunk, dtype=np.int16).tolist()
         self.audio_buffer.extend(audio_frame)  # build up the buffer until it has enough samples
 
-        if len(self.audio_buffer) >= 1280:
-            # Get prediction from openWakeWord
-            prediction = self.model.predict(self.audio_buffer)
+        while len(self.audio_buffer) >= 1280:

```

Comment:
Good solution, but needs a stop condition (or a verifiable maximum buffer size that can’t be missed in a refactor)

Reply:
In practice, I think anytime you are dropping audio you are losing performance.  @JarbasAl mentioned using a cyclical buffer but if that's implemented, there should be info logged that there is a performance issue and the plugin may not be suitable for the platform.

Reply:
Also, to enable speex noise suppression that's supported by openWakeWord, you cannot pass a list to the model, but rather an array.  I used 
`prediction = self.model.predict(np.asarray(self.audio_buffer[:1280], dtype=np.int16))` 
but I'm not sure that's efficient.